api/test/e2e/upstream_test.go

```diff
@@ -188,7 +188,60 @@ func TestRoute_Node_Host(t *testing.T) {
 	}
 }
 
-//TODO cHash
+func TestUpstream_cHash(t *testing.T) {
+	tests := []HttpTestCase{
+		{
+			caseDesc: "create cHash upstream",
+			Object:   MangerApiExpect(t),
+			Method:   http.MethodPut,
+			Path:     "/apisix/admin/upstreams/1",
+			Body: `{
+                "nodes": [{
+                    "host": "172.16.238.20",
+                    "port": 1980,
+                    "weight": 1
+				},
+				{
+                    "host": "172.16.238.20",
+                    "port": 1981,
+                    "weight": 2
+                }],
+				"type": "chash",
+				"hash_on":"header",
+				"key": "remote_addr"
+			}`,
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusOK,
+		},
+		{
+			caseDesc: "create route using the upstream just created",
+			Object:   MangerApiExpect(t),
+			Method:   http.MethodPut,
+			Path:     "/apisix/admin/routes/1",
+			Body: `{
+				"uri": "/server_port",
+				"upstream_id": "1"
+			}`,
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusOK,
+			Sleep:        sleepTime,
+		},
+		{
+			caseDesc:     "hit the route just created",
+			Object:       APISIXExpect(t),
+			Method:       http.MethodGet,
+			Path:         "/server_port",
+			ExpectStatus: http.StatusOK,
+			ExpectBody:   "1981",
+			Sleep:        sleepTime,
+		},
```

Comment:
delete upstream object

Reply:
so far, delete the upstream operation is completed in the last test case `TestUpstream_Delete`